src/chat/chatter.py

```diff
@@ -240,8 +240,11 @@ def updateAvatar(self):
             avatarPix = util.respix(url)
 
             if avatarPix:
-                self.avatarItem.setIcon(QtGui.QIcon(avatarPix))
-                self.avatarItem.setToolTip(self.avatarTip)
+                try:
+                    self.avatarItem.setIcon(QtGui.QIcon(avatarPix))
+                    self.avatarItem.setToolTip(self.avatarTip)
+                except:
```

Comment:
Catch-alls are bad. Try using QObject's destroyed() signal to remember that the qt object was destroyed, then check for that (add comments why it's needed, too).

Reply:
if I knew that I would have used it.
but Pycharm doesn't like:
except QtCore.QObject.destroyed()  [self unfilled]
except QtCore.QObject.destroyed(self)  [wrong type]
so what should it be?

yes, missed one vvv

Reply:
destroyed() is a signal, not an exception type, RTFM. Connect to the signal in __init__, set some 'we were destroyed underneath' flag in the callback, then check it here.